arch/lkl/kernel/threads.c

```diff
@@ -130,6 +133,19 @@ void free_thread_stack(struct task_struct *tsk)
 		test_tsk_thread_flag(tsk, TIF_SIGPENDING), ti, current->comm);
 
 	kill_thread(ti);
+	
+	/* free any linked list of pending signals */
+	/* note that we do not need to take the lock as the thread is dead and ought to be unreachable*/
+
+    spin_lock(&ti->signal_list_lock);
+	struct ksignal_list_node* signal_list = ti->signal_list_head;
+	while (signal_list != NULL) {
+		struct ksignal_list_node *next = signal_list->next;
+		kfree(signal_list);
+		signal_list = next;
+	}
+    spin_unlock(&ti->signal_list_lock);
```

Comment:
Indenting.  You seem to be indenting some random things with spaces.  Can't your editor display tabs as a non-whitespace character?

Reply:
Fixed, also I have found the place that makes VS Code go by the existing tabs vs spaces in a file to avoid messing them up again. Should I replace all spaces with tabs in files such as signal.c which starts out with spaces?

Actually, having a quick grep it seems only signal.c uses spaces. I'd rather not obscure the changes I am making by touching every line in this PR.

Reply:
Linux style is to use tabs for indenting, ts=8, and tabs and spaces for alignment.  I think signal.c is our code and should be moved to Linux style, but in a separate whitespace-only PR.